#!/usr/bin/env python
PACKAGE = "depth2surface_normals_seg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gaussian_blur_size = 11
# parameters
gen.add("d_erosion_size",  int_t, 0, "erosion_size for depth denoising", 2,  1, 10)
gen.add("d_gaussian_blur_size",  int_t, 0, "gaussian_blur_size for depth denoising(only Odd values)", 11,  1, 111)
gen.add("iLowH", int_t, 0, "iLowH", 30, 0, 255)
gen.add("iLowS", int_t, 0, "iLowS", 150, 0, 255)
gen.add("iLowV", int_t, 0, "iLowV", 60,  0, 255)
gen.add("iHighH", int_t, 0, "iHighH", 110, 0, 255)
gen.add("iHighS", int_t, 0, "iHighS", 255, 0, 255)
gen.add("iHighV", int_t, 0, "iHighV", 255,  0, 255)
gen.add("seg_erosion_size",  int_t, 0, "erosion_size for ground segmentation", 5,  1, 11)

exit(gen.generate(PACKAGE, "depth2surface_normals_seg", "Denoise"))
